File: core/src/commonMain/kotlin/com/xhsdn/core/platform/LivePhotoWriter.kt
package com.xhsdn.core.platform

/**
 * 跨平台 Live Photo 合成。
 * Desktop 端 always-返回 false（不合成，触发 orchestrator 的 fallback 分支）。
 */
interface LivePhotoWriter {
    /** 返回 true 表示成功；false 表示失败或不支持，调用方应走 fallback。 */
    suspend fun create(imagePath: String, videoPath: String, outputPath: String): Boolean
}

File: web/src/platform/LivePhotoWriter.ts
/**
 * Node 版 LivePhotoWriter。
 *
 * Web 版目前不依赖 ffmpeg,所以 `create()` 永远返回 false,
 * 让 DownloadOrchestrator 走 fallback:分别保存为 jpg + mp4。
 * 后续可注入一个真正合成 mov 的实现。
 */
export interface LivePhotoWriter {
  create(imagePath: string, videoPath: string, outputPath: string): Promise<boolean>;
}

export class WebLivePhotoWriter implements LivePhotoWriter {
  async create(_imagePath: string, _videoPath: string, _outputPath: string): Promise<boolean> {
    return false;
  }
}
